src/mixxx.cpp

```diff
@@ -777,19 +777,16 @@ void MixxxMainWindow::finalize() {
     delete m_pGuiTick;
     delete m_pVisualsManager;
 
-    // Check for leaked ControlObjects and give warnings.
-    QList<QSharedPointer<ControlDoublePrivate> > leakedControls;
     QList<ConfigKey> leakedConfigKeys;
 
-    ControlDoublePrivate::getControls(&leakedControls);
-
+    // Check for leaked ControlObjects and give warnings.
+    QList<QSharedPointer<ControlDoublePrivate>> leakedControls =
+            ControlDoublePrivate::takeAllInstances();
     if (leakedControls.size() > 0) {
         qDebug() << "WARNING: The following" << leakedControls.size()
                  << "controls were leaked:";
```

Comment:
If this indicates a programming bug, can we add a debug assertion?

Reply:
I'm getting this on shutdown:

```
Debug [Main]: WARNING: The following 2 controls were leaked:
Debug [Main]: "[Library]" "key_notation" QObject(0x0)
Debug [Main]: "[Controls]" "TimeFormat" ControlObject(0x559fc15f0ae0)
Warning [Main]: ControlDoublePrivate::getControl returning NULL for ( "[Library]" , "key_notation" )
Warning [Main]: ControlDoublePrivate::getControl returning NULL for ( "[Controls]" , "TimeFormat" )
```

Reply:
> If this indicates a programming bug, can we add a debug assertion?

I hesitate to do this, because this assertion would be triggered on every shutdown while testing. Inconvenient. First those leaks should be fixed, but this is out of scope here.

Reply:
Confirmed and fixed. The detection of leaked controls on shutdown was both misplaced and implemented in a cumbersome way.

I have also added the debug assertion as suggested.